huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go

```diff
@@ -1,293 +1,350 @@
 			},
-			"create_time": {
-				Type:     schema.TypeString,
-				Computed: true,
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The time when the VPC channel was created.",
 			},
 			"status": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The current status of the VPC channel.`,
 			},
 		},
 	}
 }
 
-func buildApigVpcChannelHealthConfig(d *schema.ResourceData) (channels.VpcHealthConfig, error) {
-	conf := channels.VpcHealthConfig{
-		Protocol:          d.Get("protocol").(string),
+func buildVpcChannelHealthConfig(d *schema.ResourceData) (channels.VpcHealthConfig, error) {
+	protocol := d.Get("protocol").(string)
+	result := channels.VpcHealthConfig{
+		Protocol:          protocol,
 		Path:              d.Get("path").(string),
 		Port:              d.Get("port").(int),
 		ThresholdNormal:   d.Get("healthy_threshold").(int),
 		ThresholdAbnormal: d.Get("unhealthy_threshold").(int),
 		Timeout:           d.Get("timeout").(int),
 		TimeInterval:      d.Get("interval").(int),
 	}
-	// The parameter of http codes is required if protocol is set to http or https.
-	if val := d.Get("protocol"); val.(string) == PROTOCOL_HTTP || val.(string) == PROTOCOL_HTTPS {
-		if codes, ok := d.GetOk("http_code"); ok {
-			conf.HttpCodes = codes.(string)
-		} else {
-			return conf, fmtp.Errorf("The http code cannot be empty if protocol is http or https")
+	// The parameter of HTTP codes is required if protocol is set to 'HTTP' or 'HTTPS'.
+	if ProtocolType(protocol) == ProtocolTypeHttp || ProtocolType(protocol) == ProtocolTypeHttps {
+		codes, ok := d.GetOk("http_code")
+		if !ok {
+			return result, fmt.Errorf("the HTTP code cannot be empty if the protocol is 'HTTP' or 'HTTPS'")
 		}
+		result.HttpCodes = codes.(string)
 	}
-	return conf, nil
+	return result, nil
 }
 
-func buildApigVpcChannelMembers(d *schema.ResourceData, config *config.Config) ([]channels.MemberInfo, error) {
-	members := d.Get("members").(*schema.Set)
-	mType := d.Get("member_type").(string)
-	result := make([]channels.MemberInfo, members.Len())
+func buildVpcChannelMembers(d *schema.ResourceData, config *config.Config) ([]channels.MemberInfo, error) {
+	var (
+		members = d.Get("members").(*schema.Set)
+		mType   = MemberType(d.Get("member_type").(string))
+		result  = make([]channels.MemberInfo, members.Len())
+	)
+
 	// Since the API requires that the name must be supported when the member type is 'ecs'.
 	// It is necessary to query the ecs instance information from server to obtain the instance name.
 	// We will cancel this unreasonable parameter configuration in the future.
 	ecsClient, err := config.ComputeV1Client(config.GetRegion(d))
 	if err != nil {
-		return result, fmtp.Errorf("Error creating HuaweiCloud ECS v1 client: %s", err)
+		return result, fmt.Errorf("error creating HuaweiCloud ECS v1 client: %s", err)
```

Comment:
the error message should not contian `HuaweiCloud` if we use the standard package.